pkgs/development/tools/parsing/tree-sitter/grammar.nix

```diff
@@ -29,29 +28,37 @@ stdenv.mkDerivation {
       "${source}/${location}"
   ;
 
-  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin "-I${lib.getDev libcxx}/include/c++/v1";
   buildInputs = [ tree-sitter ];
 
   dontUnpack = true;
   dontConfigure = true;
 
   buildPhase = ''
+    CFLAGS="-I$src/src -O2"
+    CXXFLAGS="-I$src/src -O2"
     runHook preBuild
-    scanner_cc="$src/src/scanner.cc"
-    if [ ! -f "$scanner_cc" ]; then
-      scanner_cc=""
+    # When both scanner.{c,cc} exist, use the C++ one.
```

Comment:
Not against this in any way, but curious why?

Reply:
It's unlikely to happen that they both exist. But if they do, I think we should not compile&link both of them.
I just randomly prefer C++ since it might be more high-level.